pkg/awsutils/awsutils.go

```diff
@@ -890,6 +906,27 @@ func (cache *EC2InstanceMetadataCache) TagENI(eniID string, currentTags map[stri
 	})
 }
 
+func (cache *EC2InstanceMetadataCache) ValidateSecurityGroups(securityGroupIds []string) error {
+
+	_, err := cache.ec2SVC.DescribeSecurityGroups(&ec2.DescribeSecurityGroupsInput{
+		GroupIds: aws.StringSlice(securityGroupIds),
+	})
+
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case "InvalidGroupId.Malformed":
+				fallthrough
```

Comment:
Curious..Why `fallthrough` here? We're looking for error codes here and that should be an `either/or` scenario rather than an `and`, right? Is this even possible?

Reply:
I think, I followed through this example documentation on DescribeSecurityGroups - https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/ec2-example-security-groups.html

The reasoning here seems - we care about error with "Invalid Security Group Id" and show the details regarding `Malformed` or `NotFound` as wrapped error.  If the failure was due to something other than Malformed or NotFound, we still show them as wrapped error in Line 924.

Using that documentation example seems fine  to me as the context is present the wrapped error.  If you have an alternate suggestion, please share, I can try it.

Reply:
So, we're currently wrapping `err` even inside the switch (and not aerr.Message()) . What do you see when you test with an SG Id that has valid format vs an SG Id that is just malformed?

Reply:
Here is the test output for the scenarios.

```
# Malformed Security Group Error Message

$ go run ec2_describe_security_groups.go foobar
Invalid Security Group: InvalidGroupId.Malformed: Invalid id: "foobar" (expecting "sg-...")
	status code: 400, request id: d8c66d06-ccbc-443a-93fc-734a809d73fb
exit status 1

$ go run ec2_describe_security_groups.go sg-foobar
Invalid Security Group: InvalidGroupId.Malformed: Invalid id: "sg-foobar"
	status code: 400, request id: d367a2fe-f742-40d0-9885-6d07d38089f7
exit status 1

# Invalid Security Group Error Message

go run ec2_describe_security_groups.go sg-0b89b5a84e6b93f51
Invalid Security Group: InvalidGroup.NotFound: The security group 'sg-0b89b5a84e6b93f51' does not exist
	status code: 400, request id: 76efcb2b-cec0-45da-bc07-8fc9002b10d7
exit status 1

# Valid Security Group

$ go run ec2_describe_security_groups.go sg-0b89...
Security Group:
{
  Description: "Allow SSH access",
  GroupId: "sg-0b89...",
  GroupName: "eksctl-senthil-17Nov-nodegroup-standard-workers-remoteAccess",
  [],
  VpcId: "vpc-xxx"
}
```

And it was against this code.

```
 func main() {
     if len(os.Args) < 2 {
         fmt.Println("Security Group ID required\nUsage: %s group_id ...", filepath.Base(os.Args[0]))
         os.Exit(1)
     }
     groupIds := os.Args[1:]

     sess, err := session.NewSession(&aws.Config{
         Region: aws.String("us-west-2")},
     )

     svc := ec2.New(sess)
     result, err := svc.DescribeSecurityGroups(&ec2.DescribeSecurityGroupsInput{
         GroupIds: aws.StringSlice(groupIds),
     })
     if err != nil {
         if aerr, ok := err.(awserr.Error); ok {
             switch aerr.Code() {
             case "InvalidGroupId.Malformed":
                 fallthrough
             case "InvalidGroup.NotFound":
                 e := errors.Wrap(err, "Invalid Security Group")
                 fmt.Println(e)
                 os.Exit(1)
             }
         }
         e := errors.Wrap(err, "Invalid Security Group")
         fmt.Println(e)
         os.Exit(1)
     }

     fmt.Println("Security Group:")
     for _, group := range result.SecurityGroups {
         fmt.Println(group)
     }
 }
```

Reply:
@achevuru  - Here is thought, I can remove the entire switch case itself. I think, it doesn't add value other than while reading the program. I had used it from the documentation example and I think, documentation example motivation was to illustrate the different kinds of errors.   

Having it simply as 

```
     if err != nil {
         e := errors.Wrap(err, "Invalid Security Group")
         fmt.Println(e)
         os.Exit(1)
     }

``` 

will be better. What do you think?

Reply:
Above test o/p makes sense. We're not going to have both the error codes for the same call and yes I think the sample program you pointed to is just showing how to validate all the possible error codes related to Invalid SG Id. I think the above proposed modification will suffice for what we're trying to do here.

Reply:
I have addressed this change. thank you.